src/Microsoft.AspNetCore.Mvc.Core/ControllerBase.cs

```diff
@@ -202,6 +203,15 @@ public IObjectModelValidator ObjectValidator
         [NonAction]
         public virtual StatusCodeResult StatusCode(int statusCode)
             => new StatusCodeResult(statusCode);
+        
+        /// <summary>
+        /// Creates a <see cref="StatusCodeResult"/> object by specifying a <paramref name="statusCode"/>.
+        /// </summary>
+        /// <param name="statusCode">The status code to set on the response.</param>
+        /// <returns>The created <see cref="StatusCodeResult"/> object for the response.</returns>
+        [NonAction]
+        public virtual StatusCodeResult StatusCode(HttpStatusCode statusCode)
```

Comment:
Should we also have a `public virtual ObjectResult StatusCode(HttpStatusCode statusCode, object value)` overload to mirror the `int` overload?

Reply:
That's fine too.

Reply:
I'll file a bug and make it up for grabs